src/calculator/provider/CalculatorProvider.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {useState} from "react";
+import {CalculatorContext} from "../context/calculatorContext.ts";
+import {OperationType} from "../type/enum/operationType.ts";
+import {DisplayInterface} from "../type/interface/displayInterface.ts";
+
+interface Props {
+
+}
+
+const CalculatorProvider: React.FC<React.PropsWithChildren<Props>> = ({children}) => {
+    const [display, setDisplay] = useState<DisplayInterface>({
+        prev: '',
+        operation: OperationType.RESULT,
+        after: ''
+    })
+    const validateDigitLength = (digit: string) => {
+        if (digit.length > 3) {
+            alert('숫자는 한번에 최대 3자리 수까지 입력 가능합니다.');
+            return true;
+        }
+        return false;
+    }
+    const onClickDigit = (digit: string) => {
+        if (display.operation === OperationType.RESULT) {
+            if (validateDigitLength(display.prev + digit)) {
+                return;
+            }
+            setDisplay({
+                prev: display.prev + digit,
+                operation: display.operation,
+                after: display.after
+            })
+            return;
+        }
+
+        if (validateDigitLength(display.after + digit)) {
+            return;
+        }
+
+        setDisplay({
+            prev: display.prev,
+            operation: display.operation,
+            after: display.after + digit
+        })
```

Comment:
```suggestion
     if (!validateDigitLength(display.prev + input)) {
          return {
            prev: display.prev + input,
            operation: DISPLAYING_RESULT,
            after: display.after
          };
     }

    return {
      prev: display.prev,
      operation: display.operation,
      after: display.after + input
    };
```

prev 값이 검증되어야 after값이 검증되는 로직으로 보입니다. 다음과 같이 단순화시킬 수 있지 않을까요? (return undefined는 최소한으로 작성하는게 좋다고 생각합니다)

Reply:
앗 이 부분은 설계를 안 하고 시작해서 생긴 문제인데
해당 코드로는 변화가 좀 어렵습니다...
현재는 prev, operation, after를 한 state와 한 함수로 관리를 하고 있습니다.

그래서 OperationType으로 prev 입력과 after를 구분하고 있습니다.
즉 OperationType이 Result로 돼있으면 prev 값을 계산하고
Result로 돼있지 않으면 after를 계산하고 있습니다.
그래서 완전한 분리가 어려워진 구조가 됐습니다 ㅠㅠ

따라서 리뷰어님께서 작성해주신 코드로는 작성은 어려울 거 같습니다. (ex. prev.length > 3 일때 after입력이 안 되는 문제가 생깁니다)
그래서 일단은 onClickDigit 내부에서 분기처리하도록 수정하였습니다.
onClickDigit 이벤트에서 operationType에 의해 onChangePrevNumber 이나 onChangeAfterNumber를 호출하도록 수정하였습니다.